src/datasets/transformations.jl

```diff
@@ -157,6 +161,33 @@ getobs(jdata, 15) == 15
 """
 joinobs(datas...) = JoinedData(datas)
 
+"""
+    tokenize(type, input)
+
+Tokenizes an input string or stream into pieces depending on selected type. 
+"""
+
+function tokenize(type::Symbol,input)::AbstractArray{AbstractString}
+    ts = TokenBuffer(input)
+    if type === :chars
+        while !isdone(ts)
+            character(ts)
+            flush!(ts)
+        end
+    elseif type === :words
+        while !isdone(ts)
+            spaces(ts) || character(ts)
+        end
+    end
+    return ts.tokens
+end
+
+# LearnBase functions {nobs, getobs} for AbstractArray{AbstractString}
+
+LearnBase.nobs(data::AbstractArray{AbstractString}) = size(data,1)
```

Comment:
`length(data)` does the same thing as `size(data, 1)` here. We may want to look into a smarter data container eventually as special-casing a certain array type is non-optimal, but that shouldn't hold up this PR.

Reply:
Ah, also we shouldn't be implementing `getobs` for `Base` types like `AbstractArray` and `AbstractString`.